src/server/req_manager.c

```diff
@@ -1417,37 +1412,53 @@ void
 mgr_server_set(struct batch_request *preq, conn_t *conn)
 {
 	int	  bad_attr = 0;
+	svrattrl *tmp;
 	svrattrl *plist;
+	svrattrl *ulist;
+	pbs_list_head unsetlist;
 	int	  rc;
 
+	CLEAR_HEAD(unsetlist);
 
 	plist = (svrattrl *)GET_NEXT(preq->rq_ind.rq_manager.rq_attr);
 	/*Only root at server host can set server attribute "acl_roots".*/
 	while (plist) {
-		if (find_attr(svr_attr_def, plist->al_name, SRV_ATR_LAST) >= 0) {
-			if (strcasecmp(plist->al_name, ATTR_aclroot) == 0) {
-				if (!is_local_root(preq->rq_user, preq->rq_host)) {
-					reply_badattr(PBSE_ATTRRO, bad_attr, plist, preq);
-					return;
-				}
-				break;
+		if (strcasecmp(plist->al_name, ATTR_aclroot) == 0) {
+			if (!is_local_root(preq->rq_user, preq->rq_host)) {
+				reply_badattr(PBSE_ATTRRO, bad_attr, plist, preq);
+				return;
 			}
+		} else if (plist->al_atopl.value == NULL || plist->al_atopl.value[0] == '\0') {
+			tmp = (struct svrattrl *)GET_NEXT(plist->al_link);
+			delete_link(&plist->al_link);
+			append_link(&unsetlist, &plist->al_link, plist);
+			plist = tmp;
+			continue;
 		}
 		plist = (struct svrattrl *)GET_NEXT(plist->al_link);
 	}
+	
+	/* if the unsetist has attributes, call server_unset to remove them separately */
+	ulist = (svrattrl *)GET_NEXT(unsetlist);
+	if (ulist) {
+		mgr_unset_attr(server.sv_attr, svr_attr_def, SRV_ATR_LAST, ulist,
+			preq->rq_perm, &bad_attr, (void *)&server, PARENT_TYPE_SERVER, INDIRECT_RES_CHECK);
+	}
```

Comment:
Don't we need to free list ?

Reply:
just like mgr_set_attr(), mgr_unset_attr() frees each attribute inside.